src/tyracraft_game.cpp

```diff
@@ -19,7 +19,7 @@ void TyraCraftGame::init() {
 void TyraCraftGame::loop() {
   engine->renderer.beginFrame(CameraInfo3D(&camera.position, &camera.lookPos));
   printf("%i\n", engine->info.getFps());
-  stateManager.update(engine, 1 / engine->info.getFps(), &engine->pad, &camera);
+  stateManager.update(1 / static_cast<float>(engine->info.getFps()), &camera);

```

Comment:
This was the core problem.
You was expecting `float` as parameter, but `u32` was given.
`1 / engine->info.getFps()` -> `int / int` -> `int`.

In c++ if one of the numbers is `float`, the result will be `float`
so:
`1 / static_cast<float>(engine->info.getFps())` -> `int / float` -> `float`

Reply:
Okay, I'll take a deeper look into that behaviour! Thx for that!